src/dotty/tools/dotc/core/Names.scala

```diff
@@ -347,4 +347,27 @@ object Names {
       StringBuilder.newBuilder.mapResult(s => from.fromChars(s.toCharArray, 0, s.length))
     def apply(): Builder[Char, Name] = termNameBuilder
   }
+
+  implicit val NameOrdering: Ordering[Name] = new Ordering[Name] {
+    def compare(x: Name, y: Name): Int = {
+      if (x.isTermName && y.isTypeName) 1
+      else if (x.isTypeName && y.isTermName) -1
+      else if (x.start == y.start && x.length == y.length) 0
+      else {
+        val until = Math.min(x.length, y.length)
```

Comment:
Between all those tree variants I prefer syntax that `Math.min` gives, because it's more close to the notation that I am used used to in math classes.

 Additionally, both `x.length.min(y.length)` and `x.length min y.length`, make first argument look like a receiver, making it special. 

Though I'm fine to change if you insist.